src/DmxInput.cpp

```diff
@@ -10,6 +10,13 @@
 #include <irq.h>
 #include <Arduino.h> // REMOVE ME
 
+int DmxInput::_buffer_size() {
+    int num_channels = _end_channel+1;
+    return num_channels+((4-num_channels%4)%4);
```

Comment:
Is the purpose of this to 32-align the buffer in memory?

Reply:
No.
This code rounds the buffer size up to the next whole integer. This is needed because the pio is configured to push 32 bits (four channels) at once. This makes the DMA transfers more efficient.

This approach has one downside: We may not be able to receive the last few (up to 3) DMX channels. However, i believe most DMX signals will contain a multiple of four channels so in most cases we can receive everything.